src/app/data-model/DecodableList.h

```diff
@@ -148,13 +173,18 @@ class DecodableList
 template <typename X>
 CHIP_ERROR Decode(TLV::TLVReader & reader, DecodableList<X> & x)
 {
-    VerifyOrReturnError(reader.GetType() == TLV::kTLVType_Array, CHIP_ERROR_SCHEMA_MISMATCH);
```

Comment:
So this looks good to me if we undo the changes to Decode itself.  The other parts are needed to make the ClearReader() case actually behave right, and I really appreciate you catching that it didn't!

Reply:
Decode changes removed.